packages/apputils/src/widgettracker.ts

```diff
@@ -0,0 +1,366 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import { InstanceTracker } from '@jupyterlab/coreutils';
+
+import { IDisposable } from '@phosphor/disposable';
+
+import { ISignal, Signal } from '@phosphor/signaling';
+
+import { FocusTracker, Widget } from '@phosphor/widgets';
+
+/**
+ * A tracker that tracks widgets.
+ *
+ * @typeparam T - The type of widget being tracked. Defaults to `Widget`.
+ */
+export interface IWidgetTracker<T extends Widget = Widget> extends IDisposable {
```

Comment:
I don't have a particular rooting interest in the inheritance-vs-composition debate, but I was a bit surprised to see that this interface doesn't explicitly extend from `IInstanceTracker` (though it appears to do so shape-wise). Am I missing some difference between the two interfaces, other than the obvious widget-related extensions?